worker/serve.go

```diff
@@ -0,0 +1,118 @@
+package worker
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/gabriel-vasile/mimetype"
+	"github.com/gotd/contrib/http_range"
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+)
+
+type (
+	// contentReader implements io.ReadSeeker but really Seek only returns the
+	// object's size, this reader can be used to pass to http.ServeContent but
+	// only if the caller made sure to manipulate the request in such a way that
+	// the only seeks are to find out the object's size
+	contentReader struct {
+		r    io.Reader
+		size int64
+	}
+)
+
+func newContentReader(r io.Reader, obj api.Object) (io.ReadSeeker, string, error) {
+	contentType := obj.ContentType()
+	if contentType == "" {
+		header := bytes.NewBuffer(nil)
+		mtype, err := mimetype.DetectReader(io.TeeReader(r, header))
+		if err != nil {
+			return nil, "", err
+		}
+		r = io.MultiReader(header, r)
+		contentType = mtype.String()
+	}
+
+	return &contentReader{
+		r:    r,
+		size: obj.Size,
+	}, contentType, nil
+}
+
+func (cr *contentReader) Seek(offset int64, whence int) (int64, error) {
+	return cr.size, nil
```

Comment:
This should only return the size of the object if `offset == 0 && whence == io.SeekEnd`. 
If `offset == 0 && whence == io.SeekStart` it should return `0`.
From what I can tell that's the only 2 cases that `ServeContent` uses.

All other cases should probably panic or return an error.

Reply:
Hm ok. I knew that but didn't really want to be strict about it here. Probably better if we are though 👍